### PR TITLE
Grpc rule 003

### PR DESCRIPTION
**Pull request by aaronreynoza**

### CODE REVIEW

File: rule-003/.gitignore [hunk 1]
```diff
@@ -1,6 +1,4 @@
 template
 templates
+dist
 build
-node_modules
```
Comment (joeygoksu) [inline rule-003/.gitignore:4]: Why we removed this line?

File: rule-003/Dockerfile [hunk 2]
```diff
@@ -11,26 +11,24 @@ RUN addgroup -S app && adduser -S -g app app
...
 
-COPY package.json ./
-COPY yarn.lock ./
-COPY tsconfig.json ./
+COPY ./package.json ./
+COPY ./yarn.lock ./
+COPY ./tsconfig.json ./
+COPY ./global.d.ts ./
+
+# Install dependencies
```
Comment (joeygoksu) [inline rule-003/Dockerfile:30]: Can we remove this line?

**Review by johanfol — DISMISSED** [review]

File: rule-003/__tests__/unit/logic.service.test.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ),
+  );
+  return quote;
+};
+
+describe('Logic Service', () => {
+  // let logicServiceExecutePostSpy: jest.SpyInstance;
```
Comment (joeygoksu) [inline rule-003/__tests__/unit/logic.service.test.ts:19]: Comment lines should be removed

File: rule-003/__tests__/unit/logic.service.test.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      let test = false;
+      config.networkMap =
+        '{"transactions":[{"transaction_type":"pain.001.001.11","transaction_name":"CustomerCreditTransferInitiationV11","channels":[{"channel_id":"UUIDv4","channel_name":"Fraud","typologies":[{"typology_id":"UUIDv4","typology_name":"Typology_28","typology_version":"1.1","rules":[{"rule_id":"UUIDv4","rule_name":"001_Derived_account_age_payee","rule_version":"1.5"},{"rule_id":"UUIDv4","rule_name":"002_Velocity_incoming","rule_version":"1.2"},{"rule_id":"UUIDv4","rule_name":"030_New_payee_transfer","rule_version":"1.0"}]},{"typology_id":"UUIDv4","typology_name":"Typology_27","typology_version":"1.0","rules":[{"rule_id":"UUIDv4","rule_name":"018_Exceptionally_large_outgoing_transfer","rule_version":"1.0"},{"rule_id":"UUIDv4","rule_name":"030_New_payee_transfer","rule_version":"1.0"},{"rule_id":"UUIDv4","rule_name":"031_Account_drain_multiple_transactions","rule_version":"1.0"}]}]},{"channel_id":"UUIDv4","channel_name":"AML"}]},{"transaction_type":"pain.013.001.09"},{"transaction_type":"pacs.008.001.10"},{"transaction_type":"pacs.002.001.12"}]}';
+      const result = await handleTransaction(expectedReq, () => {
```
Comment (joeygoksu) [inline rule-003/__tests__/unit/logic.service.test.ts:30]: I'm not sure what are we trying to achieve here?

File: rule-003/environment.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+REDIS_AUTH: TjEmUWes67
```
Comment (joeygoksu) [inline rule-003/environment.yml:1]: It was for openfaas, I think we're not using this yml. Can we confirm that?

File: rule-003/package.json [hunk 6]
```diff
@@ -1,50 +1,67 @@
 {
-  "name": "rule-003",
+  "name": "channel-orchestrator-service",
```
Comment (joeygoksu) [inline rule-003/package.json:2]: I'm confused, don't we need to set "rule003"?

File: rule-003/package.json [hunk 7]
```diff
@@ -1,50 +1,67 @@
...
   },
   "devDependencies": {
-    "@types/arangodb": "^3.5.10",
-    "@types/axios": "^0.14.0",
-    "@types/elastic-apm-node": "^2.7.0",
-    "@types/jest": "^26.0.22",
-    "@types/koa": "^2.13.1",
-    "@types/koa-bodyparser": "^4.3.0",
-    "@types/koa-router": "^7.4.1",
+    "@types/express": "^4.17.11",
```
Comment (joeygoksu) [inline rule-003/package.json:57]: We don't need these types for rule processors.

File: rule-003/src/app.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import Koa from 'koa';
```
Comment (joeygoksu) [inline rule-003/src/app.ts:2]: We don't need REST server anymore. ref. Openfaas discussions.

**Review by joeygoksu — CHANGES_REQUESTED** [review]

File: rule-003/src/app.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    this.configureRoutes();
+    this.configureMiddlewares();
+  }
+
+  configureMiddlewares(): void {
+    // LoggerService Middleware
+    this.use(async (ctx, next) => {
+      await next();
+      const rt = ctx.response.get('X-Response-Time');
+      LoggerService.log(`${ctx.method} ${ctx.url} - ${rt}`);
```
[the remark this answers is not in the document]
Comment (joeygoksu) [reply rule-003/src/app.ts:29]: @johanfol Do we need to update tms and crsp too?

File: rule-003/src/app.ts (hunk 8 above)
Comment (aaronreynoza) [reply rule-003/src/app.ts:2]: We're not deploying on gRPC only now. that's another topic that needs to be discussed.

**Review by johanfol — APPROVED** [review]

File: rule-003/package.json [hunk 10]
```diff
@@ -1,50 +1,63 @@
 {
```
Comment (joeygoksu) [inline rule-003/package.json:1]: package-lock.json should not be in the repo. yarn.lock?

**Review by joeygoksu — CHANGES_REQUESTED** [review]

**Review by joeygoksu — APPROVED** [review]

**Comment by joeygoksu** [comment]
*Tests will be added in another PR by @aaronreynoza

**Pull request merged**